util/helm/creds.go

```diff
@@ -146,11 +147,30 @@ func (creds AzureWorkloadIdentityCreds) GetAccessToken() (string, error) {
 		return "", fmt.Errorf("failed to get Azure access token after challenge: %w", err)
 	}
 
-	// Access token has a lifetime of 3 hours
-	storeAzureToken(key, token, 2*time.Hour)
+	tokenExpiry, err := getJWTExpiry(token)
+	if err != nil {
+		tokenExpiry = time.Now()
+	}
+
+	cacheExpiry := workloadidentity.CalclulateCacheExpiryBasedOnTokenExpiry(tokenExpiry)
+	storeAzureToken(key, token, cacheExpiry)
 	return token, nil
 }
 
+func getJWTExpiry(token string) (time.Time, error) {
+	parser := jwt.NewParser()
+	claims := jwt.MapClaims{}
+	_, _, err := parser.ParseUnverified(token, claims)
+	if err != nil {
+		return time.Time{}, fmt.Errorf("failed to parse JWT: %w", err)
+	}
+	exp, ok := claims["exp"].(float64)
+	if !ok {
+		return time.Time{}, errors.New("'exp' claim not found or invalid in token")
+	}
+	return time.Unix(int64(exp), 0), nil
```

Comment:
Can you use `claims.GetExpirationTime()` instead? The jwt library seems to have a nice robust method for parsing the expiration.

Reply:
Done